src/test/java/org/apache/sling/jcr/repoinit/MixinsTest.java

```diff
@@ -0,0 +1,131 @@
+        U.assertNodeExists("/addOneMixinOnOnePath", "nt:unstructured", Collections.emptyList());
+
+        U.parseAndExecute("add mixin mix:lockable to /addOneMixinOnOnePath");
+        U.assertNodeExists("/addOneMixinOnOnePath", "nt:unstructured", Collections.singletonList("mix:lockable"));
+    }
+
+    @Test
+    public void addTwoMixinsOnTwoPaths() throws Exception {
+        U.parseAndExecute("create path /addTwoMixinsOnOnePath1(nt:unstructured)\n"
+                + "create path /addTwoMixinsOnOnePath2(nt:unstructured)");
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Collections.emptyList());
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Collections.emptyList());
+
+        U.parseAndExecute("add mixin mix:lockable,mix:referenceable to /addTwoMixinsOnOnePath1,/addTwoMixinsOnOnePath2");
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+        U.assertNodeExists("/addTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+    }
+
+    @Test
+    public void removeOneMixinFromOnePath() throws Exception {
+        U.parseAndExecute("create path /removeOneMixinOnOnePath(nt:unstructured mixin mix:lockable)");
+        U.assertNodeExists("/removeOneMixinOnOnePath", "nt:unstructured", Collections.singletonList("mix:lockable"));
+
+        U.parseAndExecute("remove mixin mix:lockable from /removeOneMixinOnOnePath");
+        U.assertNodeExists("/removeOneMixinOnOnePath", "nt:unstructured", Collections.emptyList());
+    }
+
+    @Test
+    public void removeTwoMixinsFromTwoPaths() throws Exception {
+        U.parseAndExecute("create path /removeTwoMixinsOnOnePath1(nt:unstructured mixin mix:lockable,mix:referenceable)\n"
+                + "create path /removeTwoMixinsOnOnePath2(nt:unstructured mixin mix:lockable,mix:referenceable)");
+        U.assertNodeExists("/removeTwoMixinsOnOnePath1", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+        U.assertNodeExists("/removeTwoMixinsOnOnePath2", "nt:unstructured", Arrays.asList("mix:lockable","mix:referenceable"));
+
+        U.parseAndExecute("remove mixin mix:lockable,mix:referenceable from /removeTwoMixinsOnOnePath1,/removeTwoMixinsOnOnePath2");
+        U.assertNodeExists("/removeTwoMixinsOnOnePath1", "nt:unstructured", Collections.emptyList());
+        U.assertNodeExists("/removeTwoMixinsOnOnePath2", "nt:unstructured", Collections.emptyList());
+    }
+
+    @Test
+    public void addMixinOnNotExistingPath() throws Exception {
+        // this should just log a warning and continue
+        U.parseAndExecute("add mixin mix:lockable to /addMixinOnNotExistingPath");
+        assertNodeNotExists("/addMixinOnNotExistingPath");
+    }
+
+    @Test
+    public void removeMixinFromNotExistingPath() throws Exception {
+        // this should just log a warning and continue
```

Comment:
FWIW you could use a LogCapture utility to verify logging, as done in https://github.com/apache/sling-org-apache-sling-graphql-core/blob/d430d47df5b190107d413f31ef19052aacab4d98/src/test/java/org/apache/sling/graphql/core/engine/DefaultQueryExecutorLoggingTest.java

Reply:
Thanks for the review.  I've incorporated the suggested change at: https://github.com/apache/sling-org-apache-sling-jcr-repoinit/pull/38